docs/src/tutorial.md

```diff
@@ -0,0 +1,248 @@
+# Tutorial
+
+NLPModels.jl was created for two purposes:
+
+ - Allow users to access problem databases in an unified way.
+ Mainly, this means
+ [CUTEst.jl](https://github.com/JuliaSmoothOptimizers/CUTEst.jl),
+ but it also gives access to [AMPL
+ problems](https://github.com/JuliaSmoothOptimizers/AmplNLReader.jl),
+ as well as JuMP defined problems (e.g. as in
+ [OptimizationProblems.jl](https://github.com/JuliaSmoothOptimizers/OptimizationProblems.jl)).
+ - Allows users to create their own problems in the same way.
+ This means that an optimization method designed for NLPModels is
+ interchangeable between models.
```

Comment:
How about: As a consequence, optimization methods designed according to the NLPModels API will accept NLPModels of any provenance.